### PR TITLE
checkpoint-1

### PR DESCRIPTION
**Pull request by tpilvelis-gw**
As to not loose work and to get reviewed.
current status, sends slack message upon deadletter received.

### CODE REVIEW

File: .github/workflows/cd-aws-deadletter-watcher.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'
+
+    - name: Install dependencies using pipenv
+      run: |
+        python -m pip install --upgrade pip pipenv
+        pipenv install --dev
```
Comment (Figglewatts) [inline .github/workflows/cd-aws-deadletter-watcher.yml:23]: This should be `pipenv sync --dev` in order to only sync versions from the lockfile (for a deterministic build)

File: deadletter_watcher/create_slack_block.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from deadletter_watcher.slack_blocks.deadletter import create_deadletter_slack_block
+from deadletter_watcher.slack_blocks.actions import create_actions_slack_block
+from deadletter_watcher.slack_blocks.welcome import create_welcome_slack_block
+from typing import Dict,List
+
+def create_slack_block(cluster:str, service:str, count:int, deadletters : List[Dict]) -> List[Dict]:
```
Comment (Figglewatts) [inline deadletter_watcher/create_slack_block.py:6]: Formatting for type hinting is off here - we normally do `variable_name: type` i.e. colon directly after the variable name (without space) then a space, then the type

File: deadletter_watcher/create_slack_block.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    """Create a Slack Block from deadletter information
+    Args:
+        cluster: affected cluster
+        service: affected service bus
+        count: number of deadletters in service bus
+        deadletters: List of Dicts containg deadletter information
+        Dict Keys Required
+        'message_id':str, 'tenant_name':str, 'sender':str, 'recipient':str
+    Returns:
+        Slack Block
```
Comment (Figglewatts) [inline deadletter_watcher/create_slack_block.py:16]: Slack block?

File: deadletter_watcher/slack_blocks/deadletter.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from typing import Dict
+from deadletter_watcher.slack_blocks.options import create_options_slack_block
+
+def create_deadletter_slack_block(trx_id : str, tenant : str, sender : str, receiver : str) -> Dict:
```
Comment (Figglewatts) [inline deadletter_watcher/slack_blocks/deadletter.py:4]: Docstring required

File: deadletter_watcher/slack_blocks/options.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+from typing import List, Dict
+
+def create_options_slack_block(options : List[str]) -> List[Dict]:
```
Comment (Figglewatts) [inline deadletter_watcher/slack_blocks/options.py:3]: Docstring required

File: deadletter_watcher/slack_blocks/welcome.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+def create_welcome_slack_block(cluster, service, count):
```
Comment (Figglewatts) [inline deadletter_watcher/slack_blocks/welcome.py:1]: Docstring required

File: event_trigger_deploy/__main__.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
+import pulumi
+import pulumi_azure as azure
+from pulumi_azure import monitoring
+from pulumi_azure import core
+
+APP_NAME="dead-letter-watcher"
+ENDPOINT="https://o8l52l2o29.execute-api.eu-west-2.amazonaws.com/dev/listener"
```
Comment (Figglewatts) [inline event_trigger_deploy/__main__.py:7]: Is this a public API endpoint that we want to be accessible by anyone? This is a public repo

File: event_trigger_deploy/__main__.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+# Heirarchy
+# Resource Group / Contains the Dev Service Bus Namespace
+SERVICE_BUS_RESOURCE_GROUP="rgp-weu-gwc-filetrust-dev-sb"
```
Comment (Figglewatts) [inline event_trigger_deploy/__main__.py:11]: Again, is this something we want to make public/hardcode here?

File: event_trigger_deploy/__main__.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+# Service Bus Namespace / Contains all the Queues
+SB_NAMESPACE="glasswall-sb-weu-gwc-filetrust-dev"
```
Comment (Figglewatts) [inline event_trigger_deploy/__main__.py:13]: Is this something we want to make public/hardcode here?

File: event_trigger_deploy/__main__.py [hunk 10]
```diff
@@ -0,0 +1,75 @@
+from secrets import get_secret
+import pulumi_azure as azure
+from pulumi_azure import monitoring
+from pulumi_azure import core
+
+secrets = get_secret()
```
Comment (Figglewatts) [inline event_trigger_deploy/__main__.py:6]: using AWS secrets manager in Pulumi code???
SPICY I like it

Comment (tpilvelis-gw) [reply event_trigger_deploy/__main__.py:6]: figured it would be repeatable since it would be in the other module.

**Pull request merged**